src/main/java/com/auctionapp/api/controller/AuthController.java

```diff
@@ -0,0 +1,35 @@
+package com.auctionapp.api.controller;
+
+import com.auctionapp.api.model.dto.AuthenticationResponse;
+import com.auctionapp.api.model.dto.LoginRequest;
+import com.auctionapp.api.model.dto.RegisterRequest;
+import com.auctionapp.api.service.AuthService;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    private final AuthService authService;
+
+    public AuthController(AuthService authService) {
+        this.authService = authService;
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<String> register(@RequestBody RegisterRequest registerRequest) {
```

Comment:
Each validation you wrote for the frontend should also be included here as you don't know whether the user will submit a request through the frontend. You can add preconditions as I commented on another file to check whether required fields have been provided, and add length, etc. checks here, before you forward the request on to the `authService`.